### PR TITLE
Add more git repo hosts

### PR DESCRIPTION
**Pull request by andrewpollock**
Fixes: #243

### CODE REVIEW

File: vulnfeeds/cves/versions.go
Comment (oliverchang) [inline vulnfeeds/cves/versions.go:57]: Can we make this public and merge the code you have in https://github.com/google/osv.dev/blob/4724b9229883b104ffea7fd71e49b16e3a33819b/vulnfeeds/cpp/main.go#L42, and just call this there?

File: vulnfeeds/cves/versions.go
Comment (oliverchang) [inline vulnfeeds/cves/versions.go:129]: Maybe add a TODO to support extracting a commit hash from a github pull request URL. I've seen some cases of these in CVEs.

File: vulnfeeds/cves/versions.go
Comment (oliverchang) [inline vulnfeeds/cves/versions.go:58]: Would it make more sense to have a single function for parsing a repo url, and return a struct containing both the repo and commit in separate fields?

**Comment by another-rex** [comment]
The if lines are getting quite long, can you put each && or || part on a separate line to reduce the width of the code making it easier to read?

File: vulnfeeds/cves/versions.go
Comment (andrewpollock) [reply vulnfeeds/cves/versions.go:57]: Yes, that was my intention once this has landed. I'll make them public from the get go.

File: vulnfeeds/cves/versions.go
Comment (andrewpollock) [reply vulnfeeds/cves/versions.go:58]: I debated this, but felt that for smaller units of testability to pursue the approach I took here.

File: vulnfeeds/cves/versions.go
Comment (andrewpollock) [reply vulnfeeds/cves/versions.go:129]: Ack, I've seen them. TODO added.

I suspect this will require using the GitHub API, which expands the complexity of this function from mere string manipulation.

File: vulnfeeds/cves/versions.go
Comment (oliverchang) [reply vulnfeeds/cves/versions.go:58]: You could have both: 

you could have a underlying function that does both, which could simplify the implementation. Then have 2 convenience/public functions that only do 1.

Comment (andrewpollock) [reply vulnfeeds/cves/versions.go:58]: I think we need to align on requirements and callers for these functions. 

Right now, you can feed a URL, any URL to the `Repo()` function for the purposes of getting a known Git repo from it. That URL may or may not also align with being a Git commit (think GitHub Issue, GitHub Pull Request before we're able to resolve them to commits). There's a larger set of URLs than just Git commits, basically. There are going to be URLs where `Repo()` works and `Commit()` doesn't.

Should potential callers of `Repo()` be penalised because a Git commit wasn't derivable, but a base Git repository could?

Right now it's `extractGitCommit()`'s job to return the struct and originally `repo()` and `commit()` were individually testable implementation details for how that struct got constructed.

Comment (oliverchang) [reply vulnfeeds/cves/versions.go:58]: The returned value of this combined function can simplify set `nil` on the fields it's not able to derive, which aligns with accounting for cases where `Repo()` works but `Commit()` doesn't. 

That said, there may be benefits in terms of error handling if we separate too. I don't feel strongly enough about this, so feel free to go either way for this.

File: vulnfeeds/cves/versions.go
Comment (oliverchang) [inline vulnfeeds/cves/versions.go:60]: We should just return the error instead of logging it here, and have callers handle it. Same for `Commit()`.

File: vulnfeeds/cves/versions.go
Comment (andrewpollock) [reply vulnfeeds/cves/versions.go:58]: > The returned value of this combined function can simplify set `nil` on the fields it's not able to derive, which aligns with accounting for cases where `Repo()` works but `Commit()` doesn't.

Right now, you know if `Commit()` or `Repo()` works because you get your boolean back. Alternatively, the caller has to start digging around in structs to know if it got what it wanted, which seems more complicated to me?

File: vulnfeeds/cves/versions.go
Comment (andrewpollock) [reply vulnfeeds/cves/versions.go:60]: > We should just return the error instead of logging it here, and have callers handle it. Same for `Commit()`.

If we do that, then the simplicity of having a boolean return value for things working correctly is lost.

The current implementation was inspired from [go/gotip/episodes/30](https://goto.google.com/gotip/episodes/30)

If you don't feel that this is appropriate here, I can refactor accordingly.

Comment (oliverchang) [reply vulnfeeds/cves/versions.go:60]: That example lists a motivating factor as there being only a single possible sentinel error value, which isn't very useful. In this case, we have have at least two: one where the url.Parse failed, and the other is if it's unsupported. 

I also generally like returning errors, because they can propagate useful context up the stack, and leave it up to the caller to decide when the most appropriate time to log is, as opposed to here where we just log regardless. 

Anyway, not going to split any hairs over this minor thing and happy to just leave this as is  :) This is something we can easily refactor in the future if we need to.

**Review by oliverchang — APPROVED** [review]

**Pull request merged**
